ReactSkia/sdk/CurlNetworking.cpp

```diff
@@ -319,14 +333,24 @@ bool CurlNetworking::sendRequest(shared_ptr<CurlRequest> curlRequest, folly::dyn
       RNS_LOG_ERROR ("Not supported method\n" << curlRequest->method) ;
       goto safe_return;
   }
-  curl_multi_add_handle(curlMultihandle_, curl);
-  sem_post(&networkRequestSem_);
+  {
+    std::scoped_lock lock(curlInstanceProtectorMutex_);
+    curl_multi_add_handle(curlMultihandle_, curl);
+  }
+  sem_getvalue(&networkRequestSem_, &semCount);
+  if(!semCount){
```

Comment:
//We don't have to send signal if it is already count is 1. Using this as binary sem